go/arrow/array/builder.go

```diff
@@ -158,7 +181,55 @@ func (b *builder) UnsafeAppendBoolToBitmap(isValid bool) {
 	if isValid {
 		bitutil.SetBit(b.nullBitmap.Bytes(), b.length)
 	} else {
-		b.nullN++
+		b.nulls++
 	}
 	b.length++
 }
+
+func newBuilder(mem memory.Allocator, dtype arrow.DataType) Builder {
+	// FIXME(sbinet): use a type switch on dtype instead?
+	switch dtype.ID() {
+	case arrow.NULL:
+	case arrow.BOOL:
+		return NewBooleanBuilder(mem)
+	case arrow.UINT8:
+		return NewUint8Builder(mem)
+	case arrow.INT8:
+		return NewInt8Builder(mem)
+	case arrow.UINT16:
+		return NewUint16Builder(mem)
+	case arrow.INT16:
+		return NewInt16Builder(mem)
+	case arrow.UINT32:
+		return NewUint32Builder(mem)
+	case arrow.INT32:
+		return NewInt32Builder(mem)
+	case arrow.UINT64:
+		return NewUint64Builder(mem)
+	case arrow.INT64:
+		return NewInt64Builder(mem)
+	case arrow.HALF_FLOAT:
+	case arrow.FLOAT32:
+		return NewFloat32Builder(mem)
+	case arrow.FLOAT64:
+		return NewFloat64Builder(mem)
+	case arrow.STRING:
+	case arrow.BINARY:
+	case arrow.FIXED_SIZE_BINARY:
+	case arrow.DATE32:
+	case arrow.DATE64:
+	case arrow.TIMESTAMP:
+	case arrow.TIME32:
+	case arrow.TIME64:
+	case arrow.INTERVAL:
+	case arrow.DECIMAL:
```

Comment:
WDYT about grouping unhandled cases? Ex:
```go
case arrow.STRING, arrow.BINARY, arrow.FIXED_SIZE_BINARY, 
arrow.DATE32, arrow.DATE64, arrow.TIMESTAMP, 
arrow.TIME32, arrow.TIME64, arrow.INTERVAL, 
arrow.DECIMAL:
	panic(fmt.Errorf("arrow/array: unsupported builder for %T", dtype))
```
An same for case above & below 😉

Reply:
leaving it as is would reduce the diff-noise when the support for each of these arrays lands.

but I can change to what you suggest.